test/tests/functional/pbs_cpuset.py

```diff
@@ -38,24 +38,96 @@
+        """
+        This function will check that at least 1gb of test.img
+        file which is to be stagedout is created within 10 seconds.
+        """
+        fpath = os.path.join(TEST_USER.home, "test.img")
+        cmd = ['stat', '-c', '%s', fpath]
+        fsize = 0
+        for i in range(11):
+            rc = self.du.run_cmd(hosts=self.h0, cmd=cmd,
+                                 runas=TEST_USER)
+            if rc['rc'] == 0 and len(rc['out']) == 1:
+                try:
+                    fsize = int(rc['out'][0])
+                except:
+                    pass
+            # 1073741824 == 1Gb
+            if fsize > 1073741824:
+                break
+            else:
+                time.sleep(1)
+        if fsize <= 1073741824:
+            self.fail("Failed to create 1gb file at %s" % fpath)
+
     def setUp(self):
-        if not self.mom.is_cpuset_mom():
-            self.skipTest("Tests meant to run only on cgroup cpuset system.")
         TestFunctional.setUp(self)
 
+        # skip if there are no cpuset systems in the test cluster
+        no_csetmom = True
+        for mom in self.moms.values():
+            if mom.is_cpuset_mom():
+                no_csetmom = False
+        if no_csetmom:
+            self.skipTest("Skip on cluster without cgroup cpuset system.")
+
+        # Various host names
+        self.h0 = self.moms.values()[0].shortname
+        self.h1 = self.moms.values()[1].shortname
+        self.hostA = socket.getfqdn(self.h0)
+        self.hostB = socket.getfqdn(self.h1)
+        # Various node names. First mom may or may not be a cpuset system.
+        try:
+            self.n0 = self.server.status(
+                NODE, id='%s[0]' % (self.h0))[0]['id']
+        except PbsStatusError:
+            self.n0 = self.h0
+        self.n1 = self.h1
+        self.n2 = '%s[0]' % (self.n1)
+        self.n3 = '%s[1]' % (self.n1)
+
+        # Skip if there are less than four vnodes. There should be
+        # three from cpuset system (natural + two NUMA vnodes)
+        nodeinfo = self.server.status(NODE)
+        if len(nodeinfo) < 4:
+            self.skipTest("Not enough vnodes to run the test.")
+        # skip if second mom has less than two NUMA vnodes
+        try:
+            self.server.status(NODE, id=self.n3)
+        except PbsStatusError:
+            self.skipTest("vnode %s doesn't exist on pbs server" % (self.n3))
+        # skip if vnodes are not in free state
+        for node in nodeinfo:
+            if node['state'] != 'free':
+                self.skipTest("Not all the vnodes are in free state")
+
+        self.pbs_release_nodes_cmd = os.path.join(
+            self.server.pbs_conf['PBS_EXEC'], 'bin', 'pbs_release_nodes')
+        # number of resource ncpus to request initially
+        ncpus = self.server.status(NODE, 'resources_available.ncpus',
+                                   id=self.n3)[0]
+        # request a partial amount of ncpus in self.n3
```

Comment:
I'm afraid this comment is what I had in mind.  It is clear from the code that you are requesting only half the amount of ncpus, what I don't understand, and would like to see a comment about, is why only a partial number of ncpus instead of all of them?  Is it important to the test somehow?

Reply:
There was specific reason for using half of the available ncpus on the node.

Reply:
Ah, okay.  Then since everything else looks good, I'll let this be.  Thanks!